NsqSharp/Api/NsqHttpApi.cs

```diff
@@ -179,19 +179,21 @@ protected string Get(string route)
         /// <param name="httpMethod">The HTTP method.</param>
         /// <param name="timeoutMilliseconds">The timeout in milliseconds.</param>
         /// <param name="body">The body.</param>
+        /// <param name="contentType">The content type.</param>
         /// <returns>The response from the server.</returns>
-        protected static byte[] Request(string endpoint, HttpMethod httpMethod, int timeoutMilliseconds, byte[] body = null)
+        protected static byte[] Request(string endpoint, HttpMethod httpMethod, int timeoutMilliseconds, string contentType = null, byte[] body = null)
         {
             var webRequest = (HttpWebRequest)WebRequest.Create(endpoint);
             webRequest.Proxy = WebRequest.DefaultWebProxy;
             webRequest.Method = httpMethod == HttpMethod.Post ? "POST" : "GET";
             webRequest.Timeout = timeoutMilliseconds;
             webRequest.Accept = "application/vnd.nsq; version=1.0";
             webRequest.UserAgent = string.Format("{0}/{1}", ClientInfo.ClientName, ClientInfo.Version);
+            webRequest.UseDefaultCredentials = true;
```

Comment:
1. Assuming this is used to talk to an HTTP server in front of `nsqd` would it be more useful to allow the user to specify `ICredentials`? If they want to use default credentials they can pass `CredentialCache.DefaultCredentials`, possibly to the constructor. Enabling this by default opens a bit of a security risk.
2. Again assuming we're talking to something other than `nsqd`, should we store `Cookies` and send them with the next `POST` in case we're given a lease via a cookie?